examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py

```diff
@@ -64,7 +70,39 @@ def list_variables(self) -> List[Dict[str, Any]]:
                 "Failed to fetch variables. Status code: {response.status_code}, Message: {response.text}"
             )
 
+    def configuration(self) -> Dict[str, Any]:
+        response = self.auth_backend.get_session().get(
+            f"{self.get_api_url()}/config", headers={"Accept": "application/json"}
+        )
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise DagsterError(
+                "Failed to fetch configuration. Status code: {response.status_code}, Message: {response.text}"
+            )
+
+    def is_using_sqlite_backend(self) -> bool:
+        config = self.configuration()
+        database_section = next(
+            iter(section for section in config["sections"] if section["name"] == "database")
+        )
+        sql_alchemy_conn = next(
+            iter(
+                section
+                for section in database_section["options"]
+                if section["key"] == "sql_alchemy_conn"
+            )
+        )
+        return sql_alchemy_conn["value"].startswith("sqlite")
+
     def get_migration_state(self) -> AirflowMigrationState:
+        if self.is_using_sqlite_backend():
+            local_migration_dir = os.getenv("DAGSTER_AIRLIFT_MIGRATION_STATE_DIR")
+            if local_migration_dir is None:
+                raise DagsterError(
+                    "Using sqlite backend, but missing DAGSTER_AIRLIFT_MIGRATION_STATE_DIR env var. When using sqlite backend, you must set this env var to point to the directory containing the migration state."
+                )
+            return load_migration_state_from_yaml(Path(local_migration_dir))
```

Comment:
I'm a bit concerned about the complexity of this. 

It would seem simpler and more reliable just to rely on an environment variable directly rather than go through the rest API.

Reply:
moved off of the env var model